### PR TITLE
Watchdog support for the C target

### PR DESCRIPTION
**Pull request by Benichiwa**
This implements the watchdog feature as mentioned in the discussion: "Watchdogs #1379", and is meant to replace PR #1605, which is unmergeable because of formatting issues. 

This is coauthored with @edwardalee.

### CODE REVIEW

**Review by edwardalee — APPROVED** [review]
LGTM

File: org.lflang/src/org/lflang/generator/c/CWatchdogGenerator.java [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+                    + "];\n"
+                    + "lf_mode_change_type_t _lf_"
+                    + name
+                    + "_change_type = "
+                    + (effect.getTransition() == ModeTransition.HISTORY
+                    ? "history_transition"
+                    : "reset_transition")
+                    + ";");
+          }
+          // FIXME: include error reporter
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CWatchdogGenerator.java:223]: Unaddressed FIXME.

**Review by lhstrh — APPROVED** [review]
This looks great! Only one issue; I flagged an unaddressed FIXME.

File: org.lflang/src/org/lflang/generator/c/CWatchdogGenerator.java (hunk 1 above)
Comment (edwardalee) [reply org.lflang/src/org/lflang/generator/c/CWatchdogGenerator.java]: Addressed in 68afbc24b

**Pull request merged**